### PR TITLE
Adding autoPosition option

### LINKED ISSUE #1096: JSON "position" ignores
If I force a tooltip to appear in a position in the JSON ( example: position: right ) it ignores it and it also shows it in the middle (v 3.1.0)

External JS:
```
var intro = introJs();
intro.setOptions({
        steps: [
        {
        element: '#buscador',
        intro: '<b>Descubra otros expositores</b> según el <b>tipo de producto</b>, <b>marca</b> o <b>nombre de empresa</b>.',
        position: 'right'
        },
        {
```

Result:
https://jmp.sh/QKQyOok

### PR DESCRIPTION
**Pull request by afshinm**
Fixes #1096

### CODE REVIEW

**Comment by bozdoz** [comment]
Why is package-lock changed?

File: src/core/placeTooltip.js [hunk 1]
```diff
@@ -206,7 +203,7 @@ export default function placeTooltip(
     tooltipCssClass = this._options.tooltipClass;
   }
 
-  tooltipLayer.className = `introjs-tooltip ${tooltipCssClass}`.replace(
+  tooltipLayer.className = `introjs-tooltip ${tooltipCssClass || ''}`.replace(
```
Comment (bozdoz) [inline src/core/placeTooltip.js:206]: I would prefer a simple array with filter(Boolean) and join(' '):

```js
tooltipLayer.className = [
  'introjs-tooltip',
  tooltipCssClass,
].filter(Boolean).join(' ');
```

File: tests/core/placeTooltip.test.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      top: 0,
+      left: 0
+    });
+
+    jest.spyOn(getWindowSize, 'default').mockReturnValue({
+      height: 500,
+      width: 100
+    });
+
+    jest.spyOn(Element.prototype, 'getBoundingClientRect').mockReturnValue({
```
Comment (bozdoz) [inline tests/core/placeTooltip.test.js:122]: beautiful tests

**Review by bozdoz — CHANGES_REQUESTED** [review]
I just think the className thing could get out of hand; otherwise, looks great

**Comment by afshinm** [comment]
re package-lock: not entirely sure why it has been changed. I ran npm install locally and that updated the package lock.

**Comment by bozdoz** [comment]
> re package-lock: not entirely sure why it has been changed. I ran npm install locally and that updated the package lock.

If you're doing `npm i` and not `npm ci` then you'll get an updated package-lock.  Probably fine then.

File: src/core/placeTooltip.js (hunk 1 above)
Comment (afshinm) [reply src/core/placeTooltip.js]: @bozdoz done. also added two tests for this specific line.

**Review by bozdoz — APPROVED** [review]

**Pull request merged**